app/lib/tool/utils/http.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:http/http.dart' as http;
+import 'package:http_retry/http_retry.dart';
+import 'package:meta/meta.dart';
+
+final _transientStatusCodes = {
+  // See: https://cloud.google.com/storage/docs/xml-api/reference-status
+  429,
+  500,
+  503,
+};
+
+/// Creates a HTTP client that retries transient status codes.
+http.Client httpRetryClient({
+  http.Client innerClient,
+  int retries,
+}) {
+  return RetryClient(
+    innerClient ?? http.Client(),
+    when: (r) => _transientStatusCodes.contains(r.statusCode),
+    // TOOD: Consider implementing whenError and handle DNS + handshake errors.
+    //       These are safe, retrying after partially sending data is more
+    //       sketchy, but probably safe in our application.
+    retries: retries ?? 5,
+  );
+}
+
+/// Returns an [http.Client] which sends a `Bearer` token as `Authorization`
+/// header for each request.
+http.Client httpClientWithAuthorization({
+  @required String bearerToken,
```

Comment:
Generally, it's better to take  `Future<String> Function() tokenProvider`, because tokens become invalid over time, and so you can create a function that caches the previous token and knowns when it should refresh the token...